proxy-node-shared/src/test/java/uk/gov/ida/notification/shared/proxy/MetatronProxyTest.java

```diff
@@ -0,0 +1,144 @@
+
+    @Test
+    public void shouldReturnMetatronResponse() {
+        MetatronProxy metatronProxy = new MetatronProxy(
+                jsonClient,
+                UriBuilder
+                        .fromUri(testMetatronClientRule.baseUri())
+                        .path("/metadata")
+                        .build()
+        );
+
+        CountryMetadataResponse countryMetadata = metatronProxy.getCountryMetadata(TEST_ENTITY_ID);
+
+        assertThat(countryMetadata.getSamlSigningCertX509()).isEqualTo("SAMLSIGNINGCERTX509");
+        assertThat(countryMetadata.getSamlEncryptionCertX509()).isEqualTo("SAMLENCRYPTIONCERTX509");
+        assertThat(countryMetadata.getDestination()).isEqualTo(URI.create("https://destination.gov.uk"));
+        assertThat(countryMetadata.getEntityId()).isEqualTo(TEST_ENTITY_ID);
+        assertThat(countryMetadata.getCountryCode()).isEqualTo("CC");
+    }
+
+    @Test
+    public void shouldThrowMetatronResponseExceptionIfServerIssue() {
+        MetatronProxy metatronProxy = new MetatronProxy(
+                jsonClient,
+                UriBuilder
+                        .fromUri(testMetatronServerErrorClientRule.baseUri())
+                        .path("/metadata")
+                        .build()
+        );
+
+        MetatronResponseException metatronResponseException = assertThrows(MetatronResponseException.class, () -> {
+            metatronProxy.getCountryMetadata(TEST_ENTITY_ID);
+        });
+
+        assertThat(metatronResponseException.getResponseStatus()).isEqualTo(Response.Status.INTERNAL_SERVER_ERROR);
+        assertThat(metatronResponseException.getMessage()).isEqualTo(MetatronResponseException.ERROR_MESSAGE_FORMAT, TEST_ENTITY_ID);
+    }
+
+    @Test
+    public void shouldThrowMetatronResponseExceptionIfClientIssue() {
+        MetatronProxy metatronProxy = new MetatronProxy(
+                jsonClient,
+                UriBuilder
+                        .fromUri(testMetatronClientErrorClientRule.baseUri())
+                        .path("/metadata")
+                        .build()
+        );
+
+        MetatronResponseException metatronResponseException = assertThrows(MetatronResponseException.class, () -> {
+            metatronProxy.getCountryMetadata(TEST_ENTITY_ID);
+        });
+
+        assertThat(metatronResponseException.getResponseStatus()).isEqualTo(Response.Status.BAD_REQUEST);
+        assertThat(metatronResponseException.getMessage()).isEqualTo(String.format(MetatronResponseException.ERROR_MESSAGE_FORMAT, TEST_ENTITY_ID));
+    }
+
+    private static ProxyNodeJsonClient setupJsonClient() {
+        return new ProxyNodeJsonClient(
+                new ErrorHandlingClient(ClientBuilder.newClient()),
+                new JsonResponseProcessor(new ObjectMapper()),
+                new IstioHeaderStorage()
+        );
+    }
+
+    @Path("/metadata")
+    @Produces(MediaType.APPLICATION_JSON)
+    public static class TestMetatronResourceGoodResponse {
```

Comment:
I ❤️ that these test resources are in the test class 😄

Reply:
There is probably a fancier way of doing it. This seems to work just fine though!

Reply:
At some iteration, I'll copy this and put the metatron resources and associated gubbins inside the test classes.  It seems a much smarter / easier to understand way of doing it.